### PR TITLE
New GraphQL topics for the GiftCard object

### PR DESCRIPTION
**Pull request by erikmarr**
## Purpose of this pull request

This pull request (PR) adds GraphQL documentation for the gift card query and mutations under the quote endpoint.

## Affected DevDocs pages

- https://devdocs.magento.com/guides/v2.3/graphql/reference/quote.html

## Links to Magento source code

whatsnew
GraphQL supports gift cards. Added the [giftCardAccount query](https://devdocs.magento.com/guides/v2.3/graphql/reference/quote-giftcard-account.html), [applyGiftCardToCart mutation](https://devdocs.magento.com/guides/v2.3/graphql/reference/quote-apply-giftcard.html), [removeGiftCardFromCart mutation](https://devdocs.magento.com/guides/v2.3/graphql/reference/quote-remove-giftcard.html), and [redeemGiftCardBalanceAsStoreCredit mutation](https://devdocs.magento.com/guides/v2.3/graphql/reference/redeem-giftcard-balance.html).

### CODE REVIEW

File: _includes/graphql/cart-object.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 Attribute |  Data Type | Description
 --- | --- | ---
 `applied_coupon` | [`AppliedCoupon`][AppliedCoupon] | The `AppliedCoupon` object contains the `code` text attribute, which specifies the coupon code
+`applied_gift_cards` | [`AppliedGiftCard`][AppliedGiftCard] | The `AppliedGiftCard` object contains the `code` text attribute, which specifies the gift card code. `applied_gift_cards` is a Commerce-only attribute, defined in the GiftCardAccountGraphQl module
```
Comment (cpartica) [inline _includes/graphql/cart-object.md]: We can probably say that the `AppliedGiftCard` object that for now contains only the code is made this way as a wrapper to be extensible by 3rd party that wants to add data to it.
This is not a very important piece of information, but people might wonder why do we wrap everything in objects where we could of just used a String

File: guides/v2.3/graphql/reference/quote-apply-giftcard.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+**Request**
+
+ ``` text
+mutation {
+   applyGiftCardToCart(
+      cart_id: "IeTUiU0oCXjm0uRqGCOuhQ2AuQatogjG"
+      gift_card_code: "048FQEGYUA73"
+   ) {
+    cart {
+    applied_gift_cards {
```
Comment (keharper) [inline guides/v2.3/graphql/reference/quote-apply-giftcard.md]: Indent `applied_gift_cards` by a few space. Also, `code`.  The overall alignment might be off a bit. The last } doesn't look flush left.

File: guides/v2.3/graphql/reference/quote-remove-giftcard.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+## Syntax
+
+ `mutation: removeGiftCardFromCart(input: RemoveGiftCardFromCartInput): RemoveGiftCardFromCartOutput`
+
+## Example usage
+
+ The following example removes a gift card from the cart.
+
+ **Request**
```
Comment (keharper) [inline guides/v2.3/graphql/reference/quote-remove-giftcard.md]:
```suggestion
**Request**
```

File: guides/v2.3/graphql/reference/quote-remove-giftcard.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+ ``` text
+mutation {
```
Comment (keharper) [inline guides/v2.3/graphql/reference/quote-remove-giftcard.md]: Check the alignment. `cart_id` and `gift_card_code` are definitely misaligned.

File: guides/v2.3/graphql/reference/quote-remove-giftcard.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+       "applied_gift_cards": []
+     }
+   }
+ }
+}
+```
+
+## Input attributes
+
+ The `removeGiftCardFromCart` mutation requires the `cart_id` attribute.
```
Comment (keharper) [inline guides/v2.3/graphql/reference/quote-remove-giftcard.md]:
```suggestion
The `removeGiftCardFromCart` mutation requires the `cart_id` attribute.
```

File: guides/v2.3/graphql/reference/quote-remove-giftcard.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+### removeGiftCardFromCart object {#removeGiftCardFromCart}
+
+ The `removeGiftCardFromCart` object must contain the following attributes:
```
Comment (keharper) [inline guides/v2.3/graphql/reference/quote-remove-giftcard.md]:
```suggestion
The `removeGiftCardFromCart` object must contain the following attributes:
```

File: guides/v2.3/graphql/reference/quote-remove-giftcard.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+ Attribute |  Data Type | Description
+--- | --- | ---
+`cart_id` | String! | The unique ID that identifies the customer's cart
+`gift_card_code` | String! | The gift card code
+
+## Output attributes
+
+ The `removeGiftCardFromCart` object contains the `Cart` object.
```
Comment (keharper) [inline guides/v2.3/graphql/reference/quote-remove-giftcard.md]:
```suggestion
The `removeGiftCardFromCart` object contains the `Cart` object.
```

File: guides/v2.3/graphql/reference/quote-remove-giftcard.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+ Attribute |  Data Type | Description
```
Comment (keharper) [inline guides/v2.3/graphql/reference/quote-remove-giftcard.md]:
```suggestion
Attribute |  Data Type | Description
```

File: guides/v2.3/graphql/reference/quote-remove-giftcard.md [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+--- | --- | ---
+`cart` |[ Cart!](#CartObject) | Describes the contents of the specified shopping cart
+
+### Cart object {#CartObject}
+
+ {% include graphql/cart-object.md %}
+
+ [Cart query output]({{page.baseurl}}/graphql/reference/quote.html#cart-output) provides more information about the `Cart` object.
```
Comment (keharper) [inline guides/v2.3/graphql/reference/quote-remove-giftcard.md]:
```suggestion
[Cart query output]({{page.baseurl}}/graphql/reference/quote.html#cart-output) provides more information about the `Cart` object.
```

**Comment by keharper** [comment]
I think I missed some instances of extraneous spaces at the beginning of lines.

**Comment by cpartica** [comment]
we will merge this PR once all tickets for giftcard are merged in 2.3-develop but not before

File: guides/v2.3/graphql/reference/quote-apply-giftcard.md [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+
+## Example usage
+
+ The following example adds a gift card code called `048FQEGYUA73` to a cart.
+
+**Request**
+
+ ``` text
+mutation {
+   applyGiftCardToCart(
```
Comment (danielrenaud) [inline guides/v2.3/graphql/reference/quote-apply-giftcard.md]: The input should be
```
applyGiftCardToCart( input: {
      cart_id: "IeTUiU0oCXjm0uRqGCOuhQ2AuQatogjG"
      gift_card_code: "048FQEGYUA73"
   }
) 
```

File: guides/v2.3/graphql/reference/quote-giftcard-account.md [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+
+The `giftCardAccount` query returns information for a specific gift card.
+
+## Syntax
+
+ `giftCardAccount(code: String!): GiftCardAccount`
+
+## Example usage
+
+ The following example returns information about the `00MP4BZTB4NV` gift card code.
```
Comment (danielrenaud) [inline guides/v2.3/graphql/reference/quote-giftcard-account.md]: This code should be "048FQEGYUA73" to match the examples

File: guides/v2.3/graphql/reference/quote-apply-giftcard.md (hunk 10 above)
Comment (cpartica) [reply guides/v2.3/graphql/reference/quote-apply-giftcard.md]: it is with input, we used wrappers everywhere

File: _includes/graphql/cart-object.md (hunk 1 above)
Comment (erikmarr) [reply _includes/graphql/cart-object.md]: After talking to @keharper we decided this is a bigger issue that warrants a new topic. I created the following ticket to track the work needed:
https://jira.corp.magento.com/browse/MAGEDOC-3855

File: _data/toc/graphql.yml [hunk 12]
```diff
@@ -138,6 +138,9 @@ pages:
                 - label: updateCartItems mutation
                   url: /graphql/reference/quote-update-cart-items.html
 
+            - label: redeemGiftCardBalanceAsStoreCredit mutation
```
Comment (erikmarr) [inline _data/toc/graphql.yml]: Should this be a child of the quote section?

Comment (keharper) [reply _data/toc/graphql.yml]: No, it's not attached to a cart at all.

**Pull request merged**